pipelines/utils/metadata/utils.py

```diff
@@ -594,48 +413,15 @@ def create_update(
                 }}
             """
 
-    if api_mode == "prod":
-        url = "https://api.basedosdados.org/api/v1/graphql"
-    elif api_mode == "staging":
-        url = "https://staging.api.basedosdados.org/api/v1/graphql"
-
-    if update is True and id is not None:
+    if update is True and not isinstance(id, type(None)):
+        print(f'update {query_parameters}')
         mutation_parameters["id"] = id
-        r = requests.post(
-            url=url,
-            json={"query": query, "variables": {"input": mutation_parameters}},
-            headers=header,
-        ).json()
-
-    if update is False:
-        r = requests.post(
-            url=url,
-            json={"query": query, "variables": {"input": mutation_parameters}},
-            headers=header,
-        ).json()
-
-    r["r"] = "mutation"
-    if "data" in r and r is not None:
-        if r.get("data", {}).get(mutation_class, {}).get("errors", []) != []:
-            print(f"create: not found {mutation_class}", mutation_parameters)
-            print("create: error\n", json.dumps(r, indent=4, ensure_ascii=False), "\n")
-            id = None
-            raise Exception("create: Error")
-        else:
-            id = r["data"][mutation_class][_classe]["id"]
-            id = id.split(":")[1]
-
-            return r, id
-    else:
-        print("\n", "create: query\n", query, "\n")
-        print(
-            "create: input\n",
-            json.dumps(mutation_parameters, indent=4, ensure_ascii=False),
-            "\n",
-        )
-        print("create: error\n", json.dumps(r, indent=4, ensure_ascii=False), "\n")
-        raise Exception("create: Error")
 
+    response = backend._execute_query(query,{"input": mutation_parameters}, headers=get_headers(backend))
```

Comment:
dúvida: só precisa passar o header quando for fazer mutation? para as queries não precisa né?

Reply:
uhum! acho que eu só deixei, como nao quebrou segui hehe